typings/single-spa.d.ts

```diff
@@ -0,0 +1,99 @@
+declare module "single-spa" {
+  // ./start.js
+  export function start(): void;
+  export function isStarted(): boolean;
+
+  // ./jquery-support.js
+  export function ensureJQuerySupport(jQuery?: JQuery): void;
+
+  // ./applications/timeouts.js
+  export function setBootstrapMaxTime(
+    time: number,
+    dieOnTimeout?: boolean
+  ): void;
+  export function setMountMaxTime(time: number, dieOnTimeout?: boolean): void;
+  export function setUnmountMaxTime(time: number, dieOnTimeout?: boolean): void;
+  export function setUnloadMaxTime(time: number, dieOnTimeout?: boolean): void;
+
+  type timeoutConfig = {
+    millis: number;
+    dieOnTimeout: boolean;
+  };
+
+  type globalTimeoutConfig = {
+    bootstrap: timeoutConfig;
+    mount: timeoutConfig;
+    unmount: timeoutConfig;
+    unload: timeoutConfig;
+  };
+
+  export function reasonableTime(
+    promise: Promise<any>,
+    description: string,
+    timeoutConfig: timeoutConfig
+  ): void;
+
+  export function ensureValidAppTimeouts(
+    timeouts?: globalTimeoutConfig
+  ): globalTimeoutConfig;
+
+  // ./applications/apps.js
+  export function registerApplication(
+    appName: string,
+    applicationOrLoadingFn: Function | Promise<any>,
+    activityFn: (location: Location) => boolean,
+    customProps?: Object
+  ): void;
+
+  export function getMountedApps(): string[];
+
+  declare enum AppStatus {
+    NOT_LOADED = "NOT_LOADED",
+    LOADING_SOURCE_CODE = "LOADING_SOURCE_CODE",
+    NOT_BOOTSTRAPPED = "NOT_BOOTSTRAPPED",
+    BOOTSTRAPPING = "BOOTSTRAPPING",
+    NOT_MOUNTED = "NOT_MOUNTED",
+    MOUNTING = "MOUNTING",
+    MOUNTED = "MOUNTED",
+    UPDATING = "UPDATING",
+    UNMOUNTING = "UNMOUNTING",
+    UNLOADING = "UNLOADING",
+    SKIP_BECAUSE_BROKEN = "SKIP_BECAUSE_BROKEN"
+  }
+
+  export function getAppStatus(appName: string): AppStatus | null;
+
+  export function unloadApplication(
+    appName: string,
+    opts?: { waitForUnmount: boolean }
+  ): Promise<any>;
+
+  export function checkActivityFunctions(location: Location): string[];
+  export function getAppNames(): string[];
+
+  // ./navigation/navigation-events.js'
+  export function navigateToUrl(
+    obj:
+      | string
+      | {
+          currentTarget: {
+            href: string;
+          };
+          preventDefault: any;
+        },
+    opts?: Object
+  ): void;
+
+  // './navigation/reroute.js'
+  export function triggerAppChange(
+    pendingPromises?: any[],
```

Comment:
It's actually only accidental that you can pass in pendingPromises and eventArguments. The desired API was to allow for no arguments at all. So I would prefer to remove this from the typescript definition, and maybe update the code, too, so that people can't pass in pendingPromises and eventArguments.

Reply:
I have changed the code to not allow for pendingPromises and eventArguments in #302

Reply:
Makes sense, updated